go/cmd/dolt/commands/sqlserver/yaml_config.go

```diff
@@ -368,44 +376,62 @@ func (cfg YAMLConfig) QueryParallelism() int {
 	return *cfg.PerformanceConfig.QueryParallelism
 }
 
+// TLSKey returns a path to the servers PEM-encoded private TLS key. "" if there is none.
 func (cfg YAMLConfig) TLSKey() string {
 	if cfg.ListenerConfig.TLSKey == nil {
 		return ""
 	}
 	return *cfg.ListenerConfig.TLSKey
 }
 
+// TLSCert returns a path to the servers PEM-encoded TLS certificate chain. "" if there is none.
 func (cfg YAMLConfig) TLSCert() string {
 	if cfg.ListenerConfig.TLSCert == nil {
 		return ""
 	}
 	return *cfg.ListenerConfig.TLSCert
 }
 
+// RequireSecureTransport is true if the server should reject non-TLS connections.
 func (cfg YAMLConfig) RequireSecureTransport() bool {
 	if cfg.ListenerConfig.RequireSecureTransport == nil {
 		return false
 	}
 	return *cfg.ListenerConfig.RequireSecureTransport
 }
 
+// PersistenceBehavior is "load" if we include persisted system globals on server init
 func (cfg YAMLConfig) PersistenceBehavior() string {
 	if cfg.BehaviorConfig.PersistenceBehavior == nil {
 		return loadPerisistentGlobals
 	}
 	return *cfg.BehaviorConfig.PersistenceBehavior
 }
 
+// DataDir is the path to a directory to use as the data dir, both to create new databases and locate existing ones.
 func (cfg YAMLConfig) DataDir() string {
 	if cfg.DataDirStr != nil {
 		return *cfg.DataDirStr
 	}
 	return defaultDataDir
 }
 
+// CfgDir is the path to a directory to use to store the dolt configuration files.
 func (cfg YAMLConfig) CfgDir() string {
 	if cfg.CfgDirStr != nil {
 		return *cfg.CfgDirStr
 	}
 	return filepath.Join(cfg.DataDir(), defaultCfgDir)
 }
+
+// Socket is a path to the unix socket file
+func (cfg YAMLConfig) Socket() string {
```

Comment:
It would be good to have a test of this option from a YAML config file, but we can look into that as a follow-up.

Reply:
I'll add this to my note, so I can add them when I come back

Reply:
I just added one :)